self-tests.yml

```diff
@@ -158,33 +146,30 @@ functions:
     params:
       directory: gopath/src/github.com/evergreen-ci/evergreen
   run-make:
-    command: shell.exec
+    command: subprocess.exec
     params:
-       shell: bash
-       script: |
-         set -o xtrace
-         set -o errexit
-
-         # Configure a local go path for this build.
-         export GOPATH=`pwd`/gopath
-
-         # Set the path to nodejs binaries
-         export PATH=/opt/node/bin:gopath/src/github.com/evergreen-ci/evergreen/public/node_modules/.bin:$PATH
-
-         cd gopath/src/github.com/evergreen-ci/evergreen
-
-         # configure path to the settings file
-         export SETTINGS_OVERRIDE=`pwd`/creds.yml
-
-         # on windows we need to turn the slashes the other way
-         if [ "Windows_NT" == "$OS" ]; then
-            export GOPATH=$(cygpath -m $GOPATH)
-            export SETTINGS_OVERRIDE=$(cygpath -m $SETTINGS_OVERRIDE)
-         fi
-
-         # Run make, called with proper environment variables set,
-         # running the target.
-         ${build_env} make ${make_args|} ${target}
+      working_dir: gopath/src/github.com/evergreen-ci/evergreen
+      env:
+        AWS_KEY: ${aws_key}
+        AWS_SECRET: ${aws_secret}
+        DEBUG_ENABLED: ${debug}
+        DISABLE_COVERAGE: ${disable_coverage}
+        EVERGREEN_ALL: "true"
+        GOARCH: ${goarch}
+        GO_BIN_PATH: ${gobin}
+        GOOS: ${goos}
+        GOPATH: ${workdir}/gopath
+        GOROOT: ${goroot}
+        KARMA_REPORTER: junit
+        RACE_ENABLED: ${race_enabed}
+        SETTINGS_OVERRIDE: creds.yml
+        TEST_TIMEOUT: ${test_timeout}
+      command: |
+        bash -c
+        'if [ "Windows_NT" == "$OS" ]; then
+          export GOPATH=$(cygpath -m $GOPATH)
+        fi
```

Comment:
maybe we can have this be a shell script that writes a yaml file, and then we use update expansions?